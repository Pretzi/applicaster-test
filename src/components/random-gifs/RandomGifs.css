.RandomGifs__container{
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  margin-top: 30px;
  margin-bottom: 10px;
}

.RandomGifs__img {
  flex: 0 0 33%;
  height: 100px;
  background-size: cover;
}