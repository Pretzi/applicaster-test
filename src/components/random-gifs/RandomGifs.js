import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './RandomGifs.css';

class RandomGifs extends Component {
  render() {
    const randomGifs = this.props.gifs.map((gif, i) => (
      <div 
        key={i} 
        style={{backgroundImage: `url(${gif.image_url})`}}
        className="RandomGifs__img"
      />
    ))

    return (
      <div className="RandomGifs__container">
        {randomGifs}
      </div>
    );
  }
}

RandomGifs.displayName = 'RandomGifs';
RandomGifs.propTypes = {
  gifs: PropTypes.array
};

export default RandomGifs;