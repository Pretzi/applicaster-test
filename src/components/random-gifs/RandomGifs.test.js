import React from 'react';
import RandomGifs from './RandomGifs';
import { shallow } from 'enzyme';

describe('<RandomGifs />', () => {
  it('Should render empty ', () => {
    const wrapper = shallow(
      <RandomGifs gifs={[]} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render 3 images ', () => {
    const wrapper = shallow(
      <RandomGifs gifs={[
        { image_url: '1' },
        { image_url: '2' },
        { image_url: '3' }
      ]} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.RandomGifs__img').length).toBe(3);
  });
});