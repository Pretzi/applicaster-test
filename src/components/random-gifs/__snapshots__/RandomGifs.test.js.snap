// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RandomGifs /> Should render 3 images  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RandomGifs
    gifs={
        Array [
            Object {
              "image_url": "1",
            },
            Object {
              "image_url": "2",
            },
            Object {
              "image_url": "3",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="RandomGifs__img"
          style={
                    Object {
                              "backgroundImage": "url(1)",
                            }
          }
/>,
        <div
          className="RandomGifs__img"
          style={
                    Object {
                              "backgroundImage": "url(2)",
                            }
          }
/>,
        <div
          className="RandomGifs__img"
          style={
                    Object {
                              "backgroundImage": "url(3)",
                            }
          }
/>,
      ],
      "className": "RandomGifs__container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "className": "RandomGifs__img",
          "style": Object {
            "backgroundImage": "url(1)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "className": "RandomGifs__img",
          "style": Object {
            "backgroundImage": "url(2)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "className": "RandomGifs__img",
          "style": Object {
            "backgroundImage": "url(3)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="RandomGifs__img"
            style={
                        Object {
                                    "backgroundImage": "url(1)",
                                  }
            }
/>,
          <div
            className="RandomGifs__img"
            style={
                        Object {
                                    "backgroundImage": "url(2)",
                                  }
            }
/>,
          <div
            className="RandomGifs__img"
            style={
                        Object {
                                    "backgroundImage": "url(3)",
                                  }
            }
/>,
        ],
        "className": "RandomGifs__container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "className": "RandomGifs__img",
            "style": Object {
              "backgroundImage": "url(1)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "className": "RandomGifs__img",
            "style": Object {
              "backgroundImage": "url(2)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "className": "RandomGifs__img",
            "style": Object {
              "backgroundImage": "url(3)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<RandomGifs /> Should render empty  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RandomGifs
    gifs={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "RandomGifs__container",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "RandomGifs__container",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
